### PR TITLE
Add notch filter

### LINKED ISSUE #701: Notch filter
Hi !

I am using the mne-bids-pipeline on my MEG dataset. I have seen that, at the moment, there is no possibility to add a notch filter in the preprocessing. I was wondering if this feature will be added to the pipeline soon and if not, if there is a possibility as a user to integrate it for our needs? I would like to remove the power line noise but I would like to avoid deleting all frequencies after 60 Hz as I might used them later when doing time-frequency analysis.

Thanks a lot!

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #701

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: mne_bids_pipeline/steps/preprocessing/_03_frequency_filter.py [hunk 1]
```diff
@@ -89,10 +116,6 @@ def filter(
         h_freq=h_freq,
         l_trans_bandwidth=l_trans_bandwidth,
         h_trans_bandwidth=h_trans_bandwidth,
-        filter_length="auto",
-        phase="zero",
-        fir_window="hamming",
-        fir_design="firwin",
```
Comment (hoechenberger) [inline mne_bids_pipeline/steps/preprocessing/_03_frequency_filter.py:95]: MNE defaults

**Comment by hoechenberger** [comment]
@agramfort @larsoner I'm not sure when I'll be able to get back to this, so please feel free to take over if you have a couple minutes to spare! Or drop a comment on whether you agree with the approach taken here (adding a new function and thus some redundancy, instead of amending the existing `filter()` function)

File: mne_bids_pipeline/_config.py [hunk 2]
```diff
@@ -683,13 +683,32 @@
 l_freq: Optional[float] = None
 """
 The low-frequency cut-off in the highpass filtering step.
-Keep it None if no highpass filtering should be applied.
+Keep it ``None`` if no highpass filtering should be applied.
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:686]:
```suggestion
Keep it `None` if no highpass filtering should be applied.
```

File: mne_bids_pipeline/_config.py [hunk 3]
```diff
@@ -683,13 +683,32 @@
...
 """
 
 h_freq: Optional[float] = 40.0
 """
 The high-frequency cut-off in the lowpass filtering step.
-Keep it None if no lowpass filtering should be applied.
+Keep it ``None`` if no lowpass filtering should be applied.
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:692]:
```suggestion
Keep it `None` if no lowpass filtering should be applied.
```

File: mne_bids_pipeline/_config.py [hunk 4]
```diff
@@ -683,13 +683,32 @@
...
+"""
+
+notch_freq: Optional[Union[float, Iterable[float]]] = None
+"""
+Notch filter frequency. More than one frequency can be supplied, e.g. to remove
+harmonics. Keep it ``None`` if no notch filter should be applied.
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:698]:
```suggestion
harmonics. Keep it `None` if no notch filter should be applied.
```

**Review by larsoner — COMMENTED** [review]
I think you just need to

1. Add to documentation (see failing CI)
2. Add to some example script (maybe increase the `l_freq` of some dataset and add a `notch_freq` to it so we can see that it works?

**Comment by hoechenberger** [comment]
Yes!! I really just pushed "what I had", it's by no means finished! Feel free to take over if you want. Otherwise I can do it tomorrow

File: mne_bids_pipeline/_config.py [hunk 5]
```diff
@@ -706,6 +725,16 @@
 parameters.
 """
 
+notch_trans_bandwidth: float = 1.0
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:728]: This is now inconsistent with the other trans bandwidth settings. Is that intentional?

Comment (larsoner) [reply mne_bids_pipeline/_config.py:728]: Yes there is no `'auto'` for `notch_filter`

https://mne.tools/dev/generated/mne.io.Raw.html#mne.io.Raw.notch_filter

Comment (hoechenberger) [reply mne_bids_pipeline/_config.py:728]: Oh, right!

**Comment by hoechenberger** [comment]
Thanks @larsoner!

**Comment by hoechenberger** [comment]
@larsoner Lets cut a new release, WDYT?

**Comment by larsoner** [comment]
I guess this would be 1.1 then, sure!

**Pull request merged**